api/src/main/java/org/openmrs/module/fhir/api/impl/PractitionerServiceImpl.java

```diff
@@ -148,4 +159,46 @@ public Practitioner getPractitioner(String id) {
 	private List<Provider> searchProvidersByQuery(String query) {
 		return Context.getProviderService().getProviders(query, null, null, null, false);
 	}
+	
+	/**
+	 * @see org.openmrs.module.fhir.api.PractitionerService#createFHIRPractitioner(String)
+	 */
+	public Practitioner createFHIRPractitioner(Practitioner practitioner) {
+		Provider provider=null;
+		// If the representation has unknown gender and the birthday is null, that means the practitioner is not associated with any person resource. Because those attributes are compulsary if there is a person.
+		if (practitioner.getGender().equalsIgnoreCase(String.valueOf(AdministrativeGenderEnum.UNKNOWN))&&practitioner.getBirthDate()==null){ 
```

Comment:
Yes @surangak  that's why actually i haven't check nulls for gender and have checked nulls for B'days. But I'll keep @harsha89 's comment as a general comment. Coz sometimes I forget it to do, when it needs :) 

@surangak we cannot create a person with bland("") gender field, I just confirmed it with the server. It throws a exception. gender, bdate, given name is mandatory for create a person. 
But we can create a gender "O" person. 

So if the post representation's bday is null, we can clearly assume that this practitioner is not associated with a person object(Earlier I thought in OpenMRS we cannot keep unknown gender persons. Thats why I have checked gender also here). Coz in OpenMRS practitioner creation window you can't specify only name attribute, if that practitioner is not associated with a person, So we can create a practitioner without a person. 

But if the representation has additional parameters, then we have to try to match that representation with already existing persons and then decide whether to create a new person or not.